app/src/main/java/org/matrix/matrixandroidsdk/activity/HomeActivity.java

```diff
@@ -257,7 +259,13 @@ public boolean onChildClick(ExpandableListView parent, View v,
                     roomId = mAdapter.getPublicRoomAt(childPosition).roomId;
                 }
 
-                CommonActivityUtils.goToRoomPage(roomId, HomeActivity.this);
+                if (Intent.ACTION_SEND.equals(getIntent().getAction()) && getIntent().getType() != null) {
```

Comment:
Made a standalone activity at first, but noticed that the room-listing would need a refactor for this at first, as currently it's pretty hardwired to the `HomeActivity`.

This was a way to at least be able to share links, and refactor the list-creation in a new PR, as it's a bigger task